core/ibc/src/adapter.rs

```diff
@@ -1 +1,57 @@
+use ibc::core::{
+    ics02_client::{
+        client_consensus::AnyConsensusState, client_state::AnyClientState, client_type::ClientType,
+    },
+    ics03_connection::connection::ConnectionEnd,
+    ics04_channel::{
+        channel::ChannelEnd,
+        commitment::{AcknowledgementCommitment, PacketCommitment},
+        packet::Sequence,
+    },
+    ics24_host::{
+        identifier::ConnectionId,
+        path::{
+            AcksPath, ChannelEndsPath, ClientConnectionsPath, ClientConsensusStatePath,
+            ClientStatePath, ClientTypePath, CommitmentsPath, ConnectionsPath, ReceiptsPath,
+            SeqAcksPath, SeqRecvsPath, SeqSendsPath,
+        },
+    },
+};
 
+pub enum StorageError {}
+
+pub trait Store<K, V> {
+    fn get(&self, key: &K) -> Option<V>;
+    fn set(&self, key: &K, value: V) -> Result<(), StorageError>;
+    fn get_all_keys(&self) -> &[K];
+}
+
+pub trait ClientTypeStore: Store<ClientTypePath, ClientType> {}
+pub trait ClientStateStore: Store<ClientStatePath, AnyClientState> {}
+pub trait ConsensusStateStore: Store<ClientConsensusStatePath, AnyConsensusState> {}
+pub trait ConnectionEndStore: Store<ConnectionsPath, ConnectionEnd> {}
+pub trait ConnectionIdsStore: Store<ClientConnectionsPath, Vec<ConnectionId>> {}
+pub trait ChannelEndStore: Store<ChannelEndsPath, ChannelEnd> {}
+pub trait SendSequenceStore: Store<SeqSendsPath, Sequence> {}
+pub trait RecvSequenceStore: Store<SeqRecvsPath, Sequence> {}
+pub trait SeqAcksPathStore: Store<SeqAcksPath, Sequence> {}
+pub trait PacketCommitmentStore: Store<CommitmentsPath, PacketCommitment> {}
+pub trait PacketReceiptsSet: Store<ReceiptsPath, ()> {}
+pub trait PacketAcsStore: Store<AcksPath, AcknowledgementCommitment> {}
+
+pub trait IbcAdapter:
+    ChannelEndStore
+    + ClientStateStore
+    + ClientTypeStore
+    + ConnectionEndStore
+    + ConnectionIdsStore
+    + ConsensusStateStore
+    + PacketCommitmentStore
+    + PacketReceiptsSet
+    + PacketAcsStore
+    + RecvSequenceStore
+    + SendSequenceStore
+    + SeqAcksPathStore
```

Comment:
I do not think it is a good idea that abstract each storage structure into a separate trait. IMO the storage schema behavior can be described as `get()`, `set()`, `remove()`, `iter() (optional)`. Different schemas just have different types and codec implementations.